editor/editor_properties.cpp

```diff
@@ -712,6 +764,14 @@ class EditorPropertyLayersGrid : public Control {
 	Vector<String> names;
 	Vector<String> tooltips;
 
+	void set_read_only(bool p_read_only) {
+		read_only = p_read_only;
+	}
+
+	bool is_read_only() {
+		return read_only;
+	}
```

Comment:
This method is not necessary, as the readonly is checked only internally in this class. Thus you could just do `if (read_only)`.